pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -469,7 +470,14 @@ func (sv openShiftClusterStaticValidator) validatePlatformWorkloadIdentityProfil
 	}
 
 	// Validate the PlatformWorkloadIdentities
+	foundIdentityResourceIDs := []string{}
+
 	for name, p := range pwip.PlatformWorkloadIdentities {
+		if slices.Contains(foundIdentityResourceIDs, p.ResourceID) {
```

Comment:
I believe, since we have seen resourceIDs allow the case insensitivity, so we should check the map/list considering the same, i.e. `resourceID` is equivalent to `ResourceID`. Disallow the resourceIDs if the case differs but they are still same.

Reply:
Not directly related to this PR, but on the topic of case sensitivity, how do we handle that for the operator names during dynamic validation? I peeked at the code out of curiosity and didn't find any case sensitivity special handling.

Reply:
I amended the commit to be case insensitive (and switched to a map instead of a slice)